pkg/component/controller/etcd.go

```diff
@@ -245,3 +252,12 @@ func (e *Etcd) Healthy() error {
 	cancel()
 	return err
 }
+
+func detectUnsupportedEtcdArch() error {
+	if strings.Contains(runtime.GOARCH, "arm") {
+		if _, set := os.LookupEnv("ETCD_UNSUPPORTED_ARCH"); !set {
```

Comment:
```suggestion
		if os.Getenv("ETCD_UNSUPPORTED_ARCH") != runtime.ARCH {
```
The current way makes the check pass in e.g. `arm64` archi bu setting up env `ETCD_UNSUPPORTED_ARCH=foo`. And that'll trip over etcd again.